packages/components/src/components/toggle/_toggle.scss

```diff
@@ -19,461 +19,586 @@
-    width: carbon--rem(48px);
-    height: carbon--rem(24px);
-
-    // Toggle background oval
-    &::before {
-      position: absolute;
-      top: 0;
-      display: block;
-      width: carbon--rem(48px);
-      height: carbon--rem(24px);
-      box-sizing: border-box;
-      background-color: $toggle-off;
-      border-radius: carbon--rem(15px);
-      // Corresponds to the double-border for focused state (`0 0 0 1px $ui-02, 0 0 0 3px $focus`)
-      box-shadow: 0 0 0 1px transparent, 0 0 0 3px transparent;
-      content: '';
-      cursor: pointer;
-      transition: box-shadow $duration--fast-01 motion(exit, productive),
-        background-color $duration--fast-01 motion(exit, productive);
-      will-change: box-shadow;
+      &:focus {
+        outline: none;
+      }
     }
 
-    // Toggle circle
-    &::after {
-      position: absolute;
-      top: carbon--rem(3px);
-      left: carbon--rem(3px);
-      display: block;
-      width: carbon--rem(18px);
-      height: carbon--rem(18px);
-      box-sizing: border-box;
-      background-color: $icon-on-color;
-      border-radius: 50%;
-      content: '';
+    .#{$prefix}--toggle__appearance {
+      display: grid;
+      align-items: center;
+      column-gap: $spacing-03;
       cursor: pointer;
-      transition: transform $duration--fast-01 motion(exit, productive);
+      grid-template-columns: max-content max-content;
     }
-  }
 
-  .#{$prefix}--toggle__check {
-    position: absolute;
-    z-index: 1;
-    top: carbon--rem(6px);
-    left: carbon--rem(6px);
-    width: carbon--rem(6px);
-    height: carbon--rem(5px);
-    fill: $icon-on-color;
-    transform: scale(0.2);
-    transition: $duration--fast-01 motion(exit, productive);
-  }
+    .#{$prefix}--toggle__switch {
+      @include high-contrast-mode('outline');
 
-  .#{$prefix}--toggle__text--left,
-  .#{$prefix}--toggle__text--right {
-    @include type-style('body-short-01');
+      position: relative;
+      width: rem(48px);
+      height: rem(24px);
+      background-color: $toggle-off;
+      border-radius: rem(12px);
+      transition: background-color $duration--fast-01 motion(exit, productive);
 
-    position: relative;
-    margin-left: $carbon--spacing-03;
-  }
+      &::before {
+        @include high-contrast-mode('outline');
+
+        position: absolute;
+        top: rem(3px);
+        left: rem(3px);
+        width: rem(18px);
+        height: rem(18px);
+        background-color: $icon-on-color;
+        border-radius: 50%;
+        content: '';
+        transition: transform $duration--fast-01 motion(exit, productive);
+      }
+    }
 
-  .#{$prefix}--toggle__text--left {
-    position: absolute;
-    left: carbon--rem(48px);
-  }
+    .#{$prefix}--toggle__button:focus
+      + .#{$prefix}--toggle__label
+      .#{$prefix}--toggle__switch,
+    .#{$prefix}--toggle__button:active
+      + .#{$prefix}--toggle__label
+      .#{$prefix}--toggle__switch {
+      @include high-contrast-mode('focus');
```

Comment:
Sorry about the HCM merge conflicts! Looks like we might have some duplicates. I think for consistency @dakahn and I were placing them at the bottom of files so that we could ensure the styles were not overridden when a user is in HCM

Reply:
Good to know! Just pushed an update moving the hcm styles to the bottom. Could you check again if it fits the convention now?

Reply:
Perfect!